TDAT: protected void tell again ( final q name message ) throws remote exception {  <NL> final abstract coord participant proxy proxy =  <NL> this . get participant proxy ( ) ;  <NL>  <NL> if ( state . message _ cancel . equals ( message ) ) {  <NL> proxy . cancel operation ( null ) ;  <NL>  <NL>  } else if ( state . message _ compensate . equals ( message ) ) {  <NL> proxy . compensate operation ( null ) ;  <NL>  <NL>  } else if ( state . message _ close . equals ( message ) ) {  <NL> proxy . close operation ( null ) ;  <NL>  <NL>  } else if ( state . message _ exited . equals ( message ) ) {  <NL> proxy . exited operation ( null ) ;  <NL>  <NL>  } else if ( state . message _ faulted . equals ( message ) ) {  <NL> proxy . faulted operation ( null ) ;  <NL>  <NL>  } else if ( state . message _ getstatus . equals ( message ) ) {  <NL> proxy . get status operation ( null ) ;  <NL>  <NL>  } else {  <NL> throw new illegal argument exception ( " sorry , cannot tell a participant " + message ) ;  <NL>  }  <NL>  }  COM: <s> forward some message to the participant </s>